### PR TITLE
VXLAN: allow multiple remotes

### PR DESCRIPTION
**Pull request by montag451**

### CODE REVIEW

File: doc/reference/network_bridge.md [hunk 1]
```diff
@@ -118,7 +118,7 @@ Key                                  | Type      | Condition             | Defau
 `tunnel.NAME.local`                  | string    | `gre` or `vxlan`      | -                         | Local address for the tunnel (not necessary for multicast `vxlan`)
 `tunnel.NAME.port`                   | integer   | `vxlan`               | `0`                       | Specific port to use for the `vxlan` tunnel
 `tunnel.NAME.protocol`               | string    | standard mode         | -                         | Tunneling protocol: `vxlan` or `gre`
-`tunnel.NAME.remote`                 | string    | `gre` or `vxlan`      | -                         | Remote address for the tunnel (not necessary for multicast `vxlan`)
+`tunnel.NAME.remote`                 | string    | `gre` or `vxlan`      | -                         | Remote address for the tunnel. For a `vxlan` tunnel it is possible to specify multiple IP addresses separated by comma (not necessary for multicast `vxlan`)
```
Comment (ru-fu) [inline doc/reference/network_bridge.md:121]:
```suggestion
`tunnel.NAME.remote`                 | string    | `gre` or `vxlan`      | -                         | Remote address for the tunnel - for a `vxlan` tunnel, you can specify multiple IP addresses separated by comma (not necessary for multicast `vxlan`)
```

File: lxd/network/driver_bridge.go [hunk 2]
```diff
@@ -1457,9 +1457,12 @@ func (n *bridge) setup(oldConfig map[string]string) error {
 				Link: ip.Link{Name: tunName},
 			}
 
+			remotes := strings.Split(tunRemote, ",")
```
Comment (tomponline) [inline lxd/network/driver_bridge.go:1460]: Please can you use `shared.SplitNTrimSpace(args[1], ",", -1, false)` here.

File: lxd/network/driver_bridge.go [hunk 3]
```diff
@@ -1502,6 +1505,20 @@ func (n *bridge) setup(oldConfig map[string]string) error {
...
 				return err
 			}
+
+			if len(remotes) > 1 {
+				for _, remote := range remotes {
+					if _, err := shared.RunCommand(
+						"bridge", "fdb",
+						"append", "00:00:00:00:00:00",
+						"dev", tunName,
+						"dst", strings.TrimSpace(remote),
```
Comment (tomponline) [inline lxd/network/driver_bridge.go:1515]: No need for this when using `shared.SplitNTrimSpace(args[1], ",", -1, false)`

**Review by tomponline — COMMENTED** [review]
Please can you explain your use case for this change and what it achieves?

Thanks

**Comment by montag451** [comment]
> Please can you explain your use case for this change and what it achieves?
> 
> Thanks

When multicasting is not possible and you have multiple nodes in a cluster, the current solution is to create a VXLAN interface for each node in the cluster which is a bit cumbersome and a waste of VXLAN IDs. Allowing multiple remotes to be specified remove the need to create multiple VXLAN interfaces.

**Comment by tomponline** [comment]
Thanks! Does this work in a LXD cluster environment where the config of `tunnel.NAME.remote` would be applied to each cluster member host? I'm wondering if this would cause an issue if an entry is added pointing back to itself?

**Comment by tomponline** [comment]
Please can you sign off your commits too.

File: lxd/network/driver_bridge.go (hunk 2 above)
Comment (montag451) [reply lxd/network/driver_bridge.go]: Done!

File: lxd/network/driver_bridge.go (hunk 3 above)
Comment (montag451) [reply lxd/network/driver_bridge.go]: Done!

**Comment by montag451** [comment]
> Please can you sign off your commits too.

Done!

**Comment by montag451** [comment]
Yes sorry I forgot to sign off one of the commit. By incomplete, you mean that it lacks my email address? Is it really necessary?

**Comment by tomponline** [comment]
It is indeed, see https://github.com/canonical/lxd/blob/main/CONTRIBUTING.md#developer-certificate-of-origin

**Comment by montag451** [comment]
OK, I will add my email address then :)

**Comment by montag451** [comment]
Done! Sorry for all the force-pushing :)

**Comment by montag451** [comment]
> Thanks! Does this work in a LXD cluster environment where the config of `tunnel.NAME.remote` would be applied to each cluster member host? I'm wondering if this would cause an issue if an entry is added pointing back to itself?

Good remark, I will check

**Comment by montag451** [comment]
I made some tests and when an entry in the forwarding table point to a local IP address, everything works fine except that some annoying messages can be generated by the kernel. For example with a VXLAN interface named `vxlan` inside a bridge named `br0` the messages generated by the kernel looks like the one below:
```
 br0: received packet on vxlan with own address as source address (addr:aa:46:8c:1e:71:0e, vlan:0)
```

**Comment by tomponline** [comment]
> I made some tests and when an entry in the forwarding table point to a local IP address, everything works fine except that some annoying messages can be generated by the kernel. For example with a VXLAN interface named `vxlan` inside a bridge named `br0` the messages generated by the kernel looks like the one below:
> 
> ```
>  br0: received packet on vxlan with own address as source address (addr:aa:46:8c:1e:71:0e, vlan:0)
> ```

Hi, thanks for looking into this.

I am concerned that this indicates a "loop" that is being created by these full mesh of tunnels, and although the bridge seems to be dealing with it, it does suggest that this setup is causing unnecessary egress traffic over the tunnels.

Although I like the idea of this, I am not sure we can merge this in its current state as it seems to be misconfiguring the network layout somehow. Perhaps using STP might be of use in these cases?

I'm sorry I've not got any further suggestions, but I've not got the time at the moment to investigate further.

**Comment by montag451** [comment]
I understand your concerns. I will close this PR as I have no time in the foreseeable future to improve it. Thx for your time and your kind review :)

**Pull request closed without merging**